### ISSUE TITLE
provide AsBind.instantiateSync

### ISSUE THREAD

**Issue by YPetremann** [torch2424/as-bind#22]
It should be really interesting to have a way of using AsBind synchronously, could simplify integration with tests and other things without the hasle of handling all the asynchronous things.

Here is what I do:
```js
// index.js
const { AsBind } = require('as-bind')
const fs = require('fs')
module.exports = AsBind.instantiate(
  fs.readFileSync(`${__dirname}/build/optimized.wasm`), {}
)
```
```js
// test/main.test.js
let ex = {
  main: require('..')
}
function generic_test (obj, key, tests) {
  return function () {
    for (let index in tests) {
      let { input, output } = tests[index]
      it(`${Number(index) + 1}) with "${JSON.stringify(input)}": should return ${output}`,
        async function () {
          if (ex.main instanceof Promise) ex.main = await ex.main
          let func = ex.main.exports[key]
          if (!func) this.skip()
          let result = func(...input)
          if (JSON.stringify(result) !== JSON.stringify(output)) { throw new Error(`found ${result} instead of ${output}`) }
        }
      )
    }
  }
}

describe('0) Add two numbers', generic_test(ex, 'add', [
  { input: [1, 2], output: 3 },
]))
```
Here is what I expect to do:
```js
// index.js
const { AsBind } = require('as-bind')
const fs = require('fs')
module.exports = AsBind.instantiateSync(
  fs.readFileSync(`${__dirname}/build/optimized.wasm`), {}
)
```
```js
// test/main.test.js
let ex = require('..')

function generic_test (func, tests) {
  return function () {
    for (let index in tests) {
      let { input, output } = tests[index]
      it(`${Number(index) + 1}) with "${JSON.stringify(input)}": should return ${output}`,
        function () {
          if (!func) this.skip()
          let result = func(...input)
          if (JSON.stringify(result) !== JSON.stringify(output)) { throw new Error(`found ${result} instead of ${output}`) }
        }
      )
    }
  }
}

describe('0) Add two numbers', generic_test(ex.add, [
  { input: [1, 2], output: 3 },
]))
```

**Comment by torch2424** [comment]
@YPetremann Ah, thank you very much for this issue!

Totally, the reason why i didn't add this, was because [instantiate is not the most efficient way of instantiation of wasm modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/instantiate). And I thought of it as a bit of "bad practice", and didn't want to encourage that. That being said, I did not think of the testing use case, and I think you bring up a very good point! So thank you for that! :smile: :+1: 

I'll try and open a PR for this sometime tommorrow-ish. I just got done with [wasm-summit](https://webassembly-summit.org/), so I'll keep this in a open tab. Feel free to open a PR, just let me know if you start it, but no worries I'm totally down to do this as well :smile: 

Thank you!
